mvcc: always fail when SplitKey = StartKey

tbg opened the pull request:
Previous code would return the descriptor's StartKey as valid SplitKey if there
was a value at StartKey. It isn't supposed to do that since the range is
already split at that point. See #5501.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/5510)

<!-- Reviewable:end -->

mberhault commented:
LGTM

tamird commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[storage/engine/mvcc_test.go, line 2183 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5510#-KDZwMFpRvI5K7NfZ7T0:-KDZwMFpRvI5K7NfZ7T1:676269983) ([raw file](https://github.com/cockroachdb/cockroach/blob/8558d7cfe3ccc3b5b864a69ea5272d01875e091e/storage/engine/mvcc_test.go#L2183)):*
:+1:

---

*[storage/engine/mvcc_test.go, line 2184 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5510#-KDZwIAywysWJvoghkhY:-KDZwIAywysWJvoghkhZ:-98051660) ([raw file](https://github.com/cockroachdb/cockroach/blob/8558d7cfe3ccc3b5b864a69ea5272d01875e091e/storage/engine/mvcc_test.go#L2184)):*
t.Errorf("%d: unexpected error: %s", i, err)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5510)*
<!-- Sent from Reviewable.io -->

tamird commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[storage/engine/mvcc_test.go, line 2184 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5510#-KDZwIAywysWJvoghkhY:-KDZwRi27seNRRug06GI:-742658478) ([raw file](https://github.com/cockroachdb/cockroach/blob/8558d7cfe3ccc3b5b864a69ea5272d01875e091e/storage/engine/mvcc_test.go#L2184)):*
err t.Errorf("%d: unexpected error: %v", i, err)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5510)*
<!-- Sent from Reviewable.io -->

tbg committed:
mvcc: always fail when SplitKey = StartKey

Previous code would return the descriptor's StartKey as valid SplitKey if there
was a value at StartKey. It isn't supposed to do that since the range is
already split at that point. See #5501.

tschottdorf commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[storage/engine/mvcc_test.go, line 2184 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5510#-KDZwIAywysWJvoghkhY:-KDZzu7WPRFyh02TDuPO:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8558d7cfe3ccc3b5b864a69ea5272d01875e091e/storage/engine/mvcc_test.go#L2184)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5510)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5510#-:-KD_-CPt5Fa5gvugDSxs:1417088721)*
<!-- Sent from Reviewable.io -->
